synapse/cortex.py

```diff
@@ -1218,6 +1227,104 @@ async def _initCoreQueues(self):
 
         self.multiqueue = await slab.getMultiQueue('cortex:queue', nexsroot=self.nexsroot)
 
+    async def _initSubscribers(self):
+        subhive = await self.hive.open(('cortex', 'storm', 'subscribers'))
+        self.subdict = await subhive.dict()
+        self.allsubscribers = set(self.subdict.get('*allqueues', default=[]))
+
+    async def _onLayerWrite(self, mesg):
+        '''
+        Called when a layer is written
+        '''
+        mesg = mesg[1]
+        layriden = mesg['layer']
+        nodeedits = mesg['edits']
+
+        for buid, form, edits in nodeedits:
+            for etyp, vals, _ in edits:
+                if etyp in (s_layer.EDIT_NODE_ADD, s_layer.EDIT_NODE_DEL):
+                    keys = [(form, )]
+                elif etyp in (s_layer.EDIT_PROP_SET, s_layer.EDIT_PROP_DEL):
+                    prop = vals[0]
+                    keys = [(form, prop)]
+                    if prop[0] == '.':
+                        keys.append((prop, ))
+                elif etyp in (s_layer.EDIT_TAG_SET, s_layer.EDIT_TAG_DEL):
+                    tag = '#' + vals[0]
+                    keys = [(form, tag), (tag, )]
+                elif etyp in (s_layer.EDIT_TAGPROP_SET, s_layer.EDIT_TAGPROP_DEL):
+                    tag = '#' + vals[0]
+                    prop = vals[1]
+                    keys = [(form, tag, prop), (tag, prop)]
+                else:  # TODO: nodedata and light edges
+                    continue
+
+                qmsg = (layriden, buid, form, etyp, vals)
+                for key in keys:
+                    queue = self.subdict.get('$'.join(key))
+                    if queue is not None:
+                        await self.multiqueue.put(queue, qmsg)
+
+    async def _notifyLayerAdd(self, layr):
+        '''
+        Inform subscribers about new layer
+        '''
+        queues = self.subdict.get('*allqueues', default=[])
+        for queue in queues:
+            msg = ('layer:add', layr.iden)
+            await self.multiqueue.put(queue, msg)
+
+    async def _notifyLayerDel(self, layr):
+        '''
+        Inform subscribers about layer deletion
+        '''
+        queues = self.subdict.get('*allqueues', default=[])
+        for queue in queues:
```

Comment:
I think you could use `self.allsubscribers` here instead of grabbing them again.